### ISSUE TITLE
expect_fun_call false positive?

### ISSUE THREAD

**Issue by afck** [rust-lang-nursery/rust-clippy#2979]
```rust
let msg = "bar";
Some("foo").expect(msg);
```

makes Clippy 0.0.212 complain:

```
warning: use of `expect` followed by a function call
 --> src/main.rs:3:17
  |
3 |     Some("foo").expect(msg);
  |                 ^^^^^^^^^^^ help: try this: `unwrap_or_else(|| panic!(msg))`
  |
  = note: #[warn(expect_fun_call)] on by default
```

even though that isn't a function call. [Playground](https://play.rust-lang.org/?gist=a0b1f4c10df5ad4092fe9a1c8f44859d&version=nightly&mode=debug&edition=2015)

**Comment by phansch** [comment]
Thanks, that's a false positive indeed. The lint should only warn about functions.

Looks like [methods.rs#L778](https://github.com/rust-lang-nursery/rust-clippy/blob/b2caf669a8d2fd0bd121c1e898fb2e185a7fa30c/clippy_lints/src/methods.rs#L778) is executed in this case, even though it's inside a `ExprKind::MethodCall` match:

```rust
match expr.node {
    hir::ExprKind::MethodCall(ref method_call, ref method_span, ref args) => {
    // ...
    lint_expect_fun_call(cx, expr, *method_span, &method_call.ident.as_str(), args);
}
```
